earthkit/transforms/tools.py

```diff
@@ -65,7 +66,17 @@ def wrapper(
 
             dataarray = dataarray.assign_coords({time_dim: time_coord})
 
-        return func(dataarray, *args, time_dim=time_dim, **kwargs)
+        result = func(dataarray, *args, time_dim=time_dim, **kwargs)
+
+        # If we want only full days then chop off the first and last day if we have time_shift
+        if (
+            remove_partial_periods
+            and time_shift is not None
```

Comment:
Should this also check for equivalently null `time_shift` values like `0`?

Reply:
I now check if time_shift is, so None, 0, and timedelta(0) all prevent trimming